agent/asm/asm.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2014-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package asm
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+	log "github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+
+	awscreds "github.com/aws/aws-sdk-go/aws/credentials"
+)
+
+// asmAuthDataValue is the schema for the SecretStringValue returned by ASM
+type asmAuthDataValue struct {
+	Username *string
+	Password *string
+}
+
+func NewASMClient(region string, credential credentials.IAMRoleCredentials) secretsmanageriface.SecretsManagerAPI {
```

Comment:
You may need to define our own ASMClient interface,  so that we can mock for the unit test.